examples/simple_demo_qml/ThreadRenderer.cpp

```diff
@@ -321,32 +321,84 @@ void TextureNode::NewTexture(int _id, const QSize &_size)
 void TextureNode::PrepareNode()
 {
     this->mutex.lock();
+    // new texture ID from render engine (Ogre)
     int newId = this->id;
-    QSize size = this->size;
+    QSize newSize = this->size;
     this->id = 0;
     this->mutex.unlock();
+
     if (newId)
     {
         delete this->texture;
         this->texture = nullptr;
         // note: include QQuickWindow::TextureHasAlphaChannel if the rendered content
         // has alpha.
+
+        GLuint texForQt = newId;
+
+        if (this->graphicsAPI == gz::rendering::GraphicsAPI::OPENGL)
+        {
+            // intermediate FBO for copying texture data
+            GLuint rhiFbo = 0;
+            // internal texture ID to hold texture data from render engine texture,
+            // in linear GL_RGB format
+            GLuint rhiId = 0;
+            // format of new texture from render engine
+            GLint newTexFormat = 0;
+
+            QOpenGLFunctions *f = QOpenGLContext::currentContext()->functions();
+            
+            // check that new texture is in sRGB
+            f->glBindTexture(GL_TEXTURE_2D, newId);
+            glGetTexLevelParameteriv(GL_TEXTURE_2D, 0, GL_TEXTURE_INTERNAL_FORMAT, &newTexFormat);
+            f->glBindTexture(GL_TEXTURE_2D, 0);
+
+            if (newTexFormat == GL_SRGB || newTexFormat == GL_SRGB8 ||
+                    newTexFormat == GL_SRGB_ALPHA || newTexFormat == GL_SRGB8_ALPHA8)
+            {
+                // get currently bound FBO so it can be restored later
+                GLint currentFbo;
+                f->glGetIntegerv(GL_FRAMEBUFFER_BINDING, &currentFbo);
+
+                // bind render engine texture to RHI FBO
+                f->glGenFramebuffers(1, &rhiFbo);
```

Comment:
this will create a new fbo every frame. So related to previous comment, we need to store this id and as member var, and here only create the frame buffer if the fbo does not exist yet, i.e. `rhiFbo` is 0. Same goes for the `glGenTextures` line below